R/epi_shift_internal.R

```diff
@@ -0,0 +1,137 @@
+#' Create a shifted predictor
+#'
+#' `step_epi_shift` creates a *specification* of a recipe step that
+#'   will add new columns of shifted data. shifted data will
+#'   by default include NA values where the shift was induced.
+#'   These can be removed with [step_naomit()], or you may
+#'   specify an alternative filler value with the `default`
+#'   argument.
+#'
+#' @param shift A vector of integers. Each specified column will be
+#'  shifted for each value in the vector.
+#' @template step-return
+#'
+#' @details The step assumes that the data are already _in the proper sequential
+#'  order_ for shifting.
+#'
+#' @family row operation steps
+#' @export
+#' @rdname step_epi_ahead
+step_epi_shift <-
+  function(recipe,
+           ...,
+           role,
+           trained,
+           shift,
+           prefix,
+           default,
+           keys,
+           columns,
+           skip,
+           id) {
+    add_step(
+      recipe,
+      step_epi_shift_new(
+        terms = dplyr::enquos(...),
+        role = role,
+        trained = trained,
+        shift = shift,
+        prefix = prefix,
+        default = default,
+        keys = keys,
+        columns = columns,
+        skip = skip,
+        id = id
+      )
+    )
+  }
+
+step_epi_shift_new <-
+  function(terms, role, trained, shift, prefix, default, keys,
+           columns, skip, id) {
+    step(
+      subclass = "epi_shift",
+      terms = terms,
+      role = role,
+      trained = trained,
+      shift = shift,
+      prefix = prefix,
+      default = default,
+      keys = keys,
+      columns = columns,
+      skip = skip,
+      id = id
+    )
+  }
+
+#' @export
+prep.step_epi_shift <- function(x, training, info = NULL, ...) {
+  step_epi_shift_new(
+    terms = x$terms,
+    role = x$role,
+    trained = TRUE,
+    shift = x$shift,
+    prefix = x$prefix,
+    default = x$default,
+    keys = x$keys,
+    columns = recipes_eval_select(x$terms, training, info),
+    skip = x$skip,
+    id = x$id
+  )
+}
+
+#' @export
+bake.step_epi_shift <- function(object, new_data, ...) {
+  if (!all(object$shift == as.integer(object$shift))) {
+    rlang::abort("step_epi_shift requires 'shift' argument to be integer valued.")
```

Comment:
The user isn't calling `step_epi_shift` so this error would be uninterpretable. It needs to report that the `ahead/lag` value should be integer valued. It's ok to have the same message for both uses, though, like "step_epi_ahead and step_epi_lag require `ahead`/`lag ..."

Reply:
Fixed it to be comprehensible.